tsdb/head.go

```diff
@@ -1853,6 +1859,33 @@ func newStripeSeries(stripeSize int, seriesCallback SeriesLifecycleCallback) *st
 	return s
 }
 
+// reset should reset the stripeSeries and appropriately trigger the associated hooks.
+// note: This method simply resets the stripeSeries struct to its initial state without
+// performing any operations on the chunks.
+func (s *stripeSeries) reset() int {
+	deletedFromPrevStripe := 0
+	totalSeries := 0
+	for i := 0; i < s.size; i++ {
+		s.locks[i].Lock()
+		deletedForCallback := make(map[chunks.HeadSeriesRef]labels.Labels, deletedFromPrevStripe)
+		for _, all := range s.hashes[i].conflicts {
+			for _, series := range all {
+				deletedForCallback[series.ref] = series.lset
+			}
+		}
+		for _, series := range s.hashes[i].unique {
+			deletedForCallback[series.ref] = series.lset
+		}
+		s.locks[i].Unlock()
+		s.hashes[i] = seriesHashmap{}
+		s.seriesLifecycleCallback.PostDeletion(deletedForCallback)
+		totalSeries += len(deletedForCallback)
+		deletedFromPrevStripe = len(deletedForCallback)
+	}
+	s.series = make([]map[chunks.HeadSeriesRef]*memSeries, s.size)
```

Comment:
This seems a no-op because it is overwritten on line 319?

Reply:
Yeah.. it is.. i just wanted to do that for correctness of the "reset" method. WDYT?

I can no do that and rename the method to something else (maybe flush? clean?)

Reply:
Updated the code:
* Renamed the method from reset to flush
* Removed the extra logic to clean the series

Reply:
Ok.. i just did that! its better indeed!